### PR TITLE
Fix #39 by porting colonel to 1.19

### LINKED ISSUE #39: 1.19 Registry sync issue on login when not present on client
This is probably a fabric api issue but I am creating this for tracking purposes

Zdc9QuJ

### PR DESCRIPTION
**Pull request by jpenilla**
Fixes #39 

Works around registry sync issues by not registering argument types with the registry, instead using our own IDs.

### CODE REVIEW

**Comment by Kas-tle** [comment]
This resolves the problem for me. Thank you for all your hard work :)

**Comment by HugPhiluu** [comment]
Resolves the problem for me as well! Thank you very much <3

File: src/main/java/net/kyori/adventure/platform/fabric/impl/ServerboundRegisteredArgumentTypesPacket.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+/**
+ * A packet sent client to server, to let the server know which optional argument types are available on the server.
+ *
+ * <p>This packet is sent by players on join, before the command tree is sent to the client.</p>
+ *
+ * @param known Known argument type ids
+ */
+public record ServerboundRegisteredArgumentTypesPacket(Set<ResourceLocation> known) {
+  public static final ResourceLocation ID = new ResourceLocation(Adventure.NAMESPACE, "registered-args");
```
Comment (zml2008) [inline src/main/java/net/kyori/adventure/platform/fabric/impl/ServerboundRegisteredArgumentTypesPacket.java]:
```suggestion
  public static final ResourceLocation ID = new ResourceLocation(Adventure.NAMESPACE, "registered_args");
```

**Review by zml2008 — CHANGES_REQUESTED** [review]

File: src/main/java/net/kyori/adventure/platform/fabric/impl/ClientboundArgumentTypeMappingsPacket.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import io.netty.buffer.Unpooled;
+import it.unimi.dsi.fastutil.ints.Int2ObjectArrayMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import net.fabricmc.fabric.api.networking.v1.PacketSender;
+import net.kyori.adventure.Adventure;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.resources.ResourceLocation;
+
+public record ClientboundArgumentTypeMappingsPacket(Int2ObjectMap<ResourceLocation> mappings) {
+  public static final ResourceLocation ID = new ResourceLocation(Adventure.NAMESPACE, "registered-arg-mappings");
```
Comment (zml2008) [inline src/main/java/net/kyori/adventure/platform/fabric/impl/ClientboundArgumentTypeMappingsPacket.java]:
```suggestion
  public static final ResourceLocation ID = new ResourceLocation(Adventure.NAMESPACE, "registered_arg_mappings");
```

**Review by zml2008 — CHANGES_REQUESTED** [review]

**Review by zml2008 — APPROVED** [review]

**Pull request merged**